agent/api/task/taskvolume.go

```diff
@@ -73,6 +75,10 @@ func (tv *TaskVolume) UnmarshalJSON(b []byte) error {
 func (tv *TaskVolume) MarshalJSON() ([]byte, error) {
 	result := make(map[string]interface{})
 
+	if len(tv.Type) == 0 {
```

Comment:
Would we require this in Marshal? since the empty volume's MarshalJSON will now contain type as "host"?

Reply:
If we have the check in unmarshalling, I don't think it's necessary in marshalling?

Reply:
yes, I agree with that this probably isn't required as it's already handled in the Marshal, but adding here to keep the consistency. I can remove that if you feel strongly about that.

Reply:
no I'm ok with double-checking.